CMakeLists.txt

```diff
@@ -32,8 +32,15 @@ project(SLEEF
 	VERSION ${SLEEF_VERSION}
 	LANGUAGES C)
 
+if(NOT DEFINED sleef_SOURCE_DIR)
```

Comment:
Two requests:

1. Please add a comment that refers to the cmake documentation that describes the auto-generated variable name.
2. Is there a way to build this variable name programmatically? What happens if we decide to change the name of the project, or a user decides to change the name of the top level folder of sleef?

Reply:
Doesn't CMAKE already have support for this with CMAKE_CURRENT_SOURCE_DIR or even CMAKE_CURRENT_LIST_DIR

It also has CMAKE_CURRENT_BINARY_DIR